tests/integration/olp-cpp-sdk-dataservice-write/VersionedLayerClientTest.cpp

```diff
@@ -0,0 +1,310 @@
+using testing::Mock;
+using testing::Return;
+namespace client = olp::client;
+namespace write = olp::dataservice::write;
+namespace model = olp::dataservice::write::model;
+
+constexpr auto kLayer = "layer";
+const std::string kPublishApiName = "publish";
+const auto kHrn = olp::client::HRN{"hrn:here:data:::catalog"};
+const auto kLookupPublishApiUrl =
+    "https://api-lookup.data.api.platform.here.com/lookup/v1/resources/" +
+    kHrn.ToString() + "/apis/publish/v2";
+const std::string kPublishUrl =
+    "https://tmp.publish.data.api.platform.here.com/publish/v2/catalogs/" +
+    kHrn.ToString() + "/publications";
+
+class VersionedLayerClientTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    network_ = std::make_shared<NetworkMock>();
+    settings_.cache =
+        olp::client::OlpClientSettingsFactory::CreateDefaultCache({});
+    settings_.network_request_handler = network_;
+    settings_.task_scheduler =
+        olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler(1);
+  }
+
+  void TearDown() override {
+    settings_.network_request_handler.reset();
+    settings_.cache.reset();
+
+    network_.reset();
+  }
+
+  write::model::Apis CreateApiResponse(const std::string& service) {
+    const auto apis =
+        mockserver::DefaultResponses::GenerateResourceApisResponse(
+            kHrn.ToCatalogHRNString());
+    auto it = std::find_if(
+        apis.begin(), apis.end(),
+        [&](const model::Api& obj) -> bool { return obj.GetApi() == service; });
+
+    write::model::Apis result;
+    if (it != apis.end()) {
+      result.push_back(*it);
+    }
+
+    return result;
+  }
+
+  std::shared_ptr<NetworkMock> network_;
+  olp::client::OlpClientSettings settings_;
+};
+
+TEST_F(VersionedLayerClientTest, StartBatch) {
+  const auto catalog = kHrn.ToCatalogHRNString();
+  const auto api = CreateApiResponse(kPublishApiName);
+  const auto publication =
+      mockserver::DefaultResponses::GeneratePublicationResponse({kLayer}, {});
+  ASSERT_FALSE(api.empty());
+
+  {
+    SCOPED_TRACE("Success request");
+
+    EXPECT_CALL(*network_, Send(IsGetRequest(kLookupPublishApiUrl), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                         olp::http::HttpStatusCode::OK),
+                                     olp::serializer::serialize(api)));
+    EXPECT_CALL(*network_, Send(IsPostRequest(kPublishUrl), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                         olp::http::HttpStatusCode::OK),
+                                     olp::serializer::serialize(publication)));
+
+    write::VersionedLayerClient write_client(kHrn, settings_);
+    const auto batch_request = model::StartBatchRequest().WithLayers({kLayer});
+    const auto response =
+        write_client.StartBatch(batch_request).GetFuture().get();
+    const auto& result = response.GetResult();
```

Comment:
Please separate a little request and waiting for the response.  Mixing it like this together makes it very hard to read.